src/Tokenizer/Analyzer/ClassyAnalyzer.php

```diff
@@ -30,7 +30,8 @@ public function isClassyInvocation(Tokens $tokens, int $index): bool
             throw new \LogicException(sprintf('No T_STRING at given index %d, got "%s".', $index, $tokens[$index]->getName()));
         }
 
-        if (\in_array(strtolower($token->getContent()), ['bool', 'float', 'int', 'iterable', 'object', 'parent', 'self', 'string', 'void', 'null', 'false', 'never'], true)) {
+        $tokenTypeAnalysis = new Analysis\TypeAnalysis($token->getContent());
+        if ($tokenTypeAnalysis->isReservedType()) {
```

Comment:
I really don't like one-time variables 😉.

```suggestion
        if ((new Analysis\TypeAnalysis($token->getContent()))->isReservedType()) {
```

PS. Static factory would be handy on `TypeAnalysis`, but it would be little weird to add it in the bug fix...